cmd/kubeadm/app/util/apiclient.go

```diff
@@ -59,22 +63,27 @@ func WaitForAPI(client *clientset.Clientset) {
 
 // WaitForPodsWithLabel will lookup pods with the given label and wait until they are all
 // reporting status as running.
-func WaitForPodsWithLabel(client *clientset.Clientset, labelKeyValPair string) {
-	// TODO: Implement a timeout
-	// TODO: Implement a verbosity switch
-	wait.PollInfinite(kubeadmconstants.APICallRetryInterval, func() (bool, error) {
+func WaitForPodsWithLabel(client clientset.Interface, timeout time.Duration, out io.Writer, labelKeyValPair string) error {
+
+	lastKnownPodNumber := -1
+	return wait.PollImmediate(kubeadmconstants.APICallRetryInterval, timeout, func() (bool, error) {
 		listOpts := metav1.ListOptions{LabelSelector: labelKeyValPair}
-		apiPods, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(listOpts)
+		pods, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(listOpts)
 		if err != nil {
-			fmt.Printf("[apiclient] Error getting Pods with label selector %q [%v]\n", labelKeyValPair, err)
+			fmt.Fprintf(out, "[apiclient] Error getting Pods with label selector %q [%v]\n", labelKeyValPair, err)
 			return false, nil
 		}
 
-		if len(apiPods.Items) == 0 {
+		if lastKnownPodNumber != len(pods.Items) {
+			fmt.Fprintf(out, "[apiclient] Found %d Pods for label selector %s\n", len(pods.Items), labelKeyValPair)
```

Comment:
I've now changed it to only report on a change; I now think they make sense.
(They didn't before indeed)